### DISCUSSION TITLE
Can database migrations be automated in production?

### DISCUSSION THREAD

**Discussion by donywan** [serverpod/serverpod#2677] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
When using Docker to package into a production image and then run it for the first time, the table will not be created.

**Describe the solution you'd like**
I hope to automatically update the database according to the migration version record when deploying the production environment.

**Describe alternatives you've considered**
I can only manually execute SQL scripts in the production database now.

**Additional context**

**Comment by SandPod** [comment, marked as answer]
Hey @donywan!

The way to achieve this currently is to add `--apply-migrations` to your docker image.
This will attempt to apply any migrations available when starting your server.

We are looking into adding a separate CLI command to run migrations but this currently does not exist.

**Comment by iampopal** [comment]
I have already tried adding `--apply-migrations` to:
FROM dart:3.3.0 AS build

```
WORKDIR /app
COPY . .

RUN dart pub get
RUN dart compile exe bin/main.dart -o bin/server

FROM alpine:latest

ENV runmode=production
ENV serverid=default
ENV logging=normal
ENV role=monolith

COPY --from=build /runtime/ /
COPY --from=build /app/bin/server server
COPY --from=build /app/config/ config/
COPY --from=build /app/web/ web/

EXPOSE 8080
EXPOSE 8081
EXPOSE 8082

ENTRYPOINT ./server --mode=$runmode --server-id=$serverid --logging=$logging --role=$role --apply-migrations
```

But it's stillnot migrating and need to be migrated manullay

**Reply by SandPod** [reply]
The reason for this is that we previosly incorrectly did not copy over the migration in the docker image.

It should work if you add this line or take inspiration from the new docker image: https://github.com/serverpod/serverpod/blob/a6ec3603305205ba19ea10f23165f27b6deaf59e/templates/serverpod_templates/projectname_server/Dockerfile#L20

**Reply by iampopal** [reply]
Thank you. By copying the migrations folder it all worked well.
